### PR TITLE
🚧Use interactor to create manual timeslot with the + button

### PR DESCRIPTION
**Pull request by sofokli1**
Doesn't close any issue

### CODE REVIEW

File: teferi/UI/Modules/Main/MainViewModel.swift [hunk 1]
```diff
@@ -222,15 +222,13 @@ class MainViewModel : RxViewModel
...
-                                             categoryWasSetByUser: true,
-                                             tryUsingLatestLocation: true)
-        else { return }
-        
-        metricsService.log(event: .timeSlotManualCreation(date: timeService.now, category: category))
-        metricsService.log(event: .timeSlotCreated(date: timeService.now, category: category, duration: nil))
+        InteractorFactory.shared.createManuallyCreateTimeSlottWithCategoryInteractor(category: category)
+            .execute()
+            .subscribe(onNext: { [unowned self] _ in
+                self.metricsService.log(event: .timeSlotManualCreation(date: self.timeService.now, category: category))
```
Comment (Odrakir) [inline teferi/UI/Modules/Main/MainViewModel.swift]: I think these two should be in the interactor as well.

File: teferi/UI/Modules/Main/MainViewModel.swift [hunk 2]
```diff
@@ -222,15 +222,13 @@ class MainViewModel : RxViewModel
...
-                                             category: category,
-                                             categoryWasSetByUser: true,
-                                             tryUsingLatestLocation: true)
-        else { return }
-        
-        metricsService.log(event: .timeSlotManualCreation(date: timeService.now, category: category))
-        metricsService.log(event: .timeSlotCreated(date: timeService.now, category: category, duration: nil))
+        InteractorFactory.shared.createManuallyCreateTimeSlottWithCategoryInteractor(category: category)
+            .execute()
+            .subscribe(onNext: { [unowned self] _ in
```
Comment (Odrakir) [inline teferi/UI/Modules/Main/MainViewModel.swift]: I'm not a big fan of subscribing in the ViewModel, I rather return an Observable from addNewSlot and subscribe from ViewController to do something after success... But let's leave it like this for now.

File: teferi/Interactors/ManuallyCreateTimeSlot.swift [hunk 3]
```diff
@@ -6,16 +6,21 @@ class ManuallyCreateTimeSlot: Interactor
...
     {
         self.repositoryService = repositoryService
         self.timeService = timeService
+        self.metricsService = metricsService
         self.category = category
     }
     
     func execute() -> Observable<Void>
     {
+        self.metricsService.log(event: .timeSlotManualCreation(date: self.timeService.now, category: self.category))
```
Comment (Odrakir) [inline teferi/Interactors/ManuallyCreateTimeSlot.swift]: What happens if the creation fails for some reason? Then we probably don't want to send metrics. This two lines should be in a side effect to the `Observable`. (`do(onNext: {...}`)

Comment (sofokli1) [reply teferi/Interactors/ManuallyCreateTimeSlot.swift]: do you mean like this?
```
func execute() -> Observable<Void>
    {
        return repositoryService.saveTimeSlot(startTime: timeService.now, category: category, edited: true)
            .do(onNext: { [unowned self] in
                self.metricsService.log(event: .timeSlotManualCreation(date: self.timeService.now, category: self.category))
                self.metricsService.log(event: .timeSlotCreated(date: self.timeService.now, category: self.category, duration: nil))
            })
    }
```

Comment (Odrakir) [reply teferi/Interactors/ManuallyCreateTimeSlot.swift]: Yes, you could even extract them to a private method, but that's the idea.

Comment (sofokli1) [reply teferi/Interactors/ManuallyCreateTimeSlot.swift]: I tried it like this before but had some issues with the `[unowned self]`

**Pull request merged**